dpx/src/dpx_pdfobj.rs

```diff
@@ -3095,38 +3096,31 @@ unsafe fn find_xref(handle: &mut InputHandleWrapper, mut file_size: i32) -> i32
  * This routine must be called with the file pointer located
  * at the start of the trailer.
  */
-unsafe fn parse_trailer(mut pf: *mut pdf_file) -> *mut pdf_obj {
+unsafe fn parse_trailer(mut pf: *mut pdf_file) -> Option<*mut pdf_obj> {
     /*
      * Fill work_buffer and hope trailer fits. This should
      * be made a bit more robust sometime.
      */
```

Comment:
The code here is still using the `static mut work_buffer_u8: [u8; 1024]`, which is unsafe as well as this concern about the trailer maybe not fitting. I don't see anything wrong with using a Vec here.

Reply:
Let's do it in another PR.
And maybe you know. Is there easy way to grow `Vec` from `Read` with exact number of bytes?

Reply:
There’s `Read::read_to_end(&mut self, &mut Vec<u8>)`.

Reply:
`read_to_end` reads to end of file, not exact number of bytes.

Reply:
maybe `file.take(5).read_to_end(` is what I need